pkg/cmd/target.go

```diff
@@ -739,6 +769,70 @@ func gardenWrapper(targetReader TargetReader, targetWriter TargetWriter, configR
 	return nil
 }
 
+func serverWrapper(reader ConfigReader, serverName string) {
+	config := reader.ReadConfig(pathGardenConfig)
+
+	if findIP(serverName) != "" {
+		for _, garden := range config.GardenClusters {
+			kc := garden.KubeConfig
+			svr := getServerValueFromKubeconfig(kc)
+			addrIP, err := net.LookupIP(removeHTTPSandPort(svr))
+			checkError(err)
+			if addrIP[0].String() == removeHTTPSandPort(serverName) {
+				pgarden = garden.Name
+			}
+		}
+	} else {
+		for _, garden := range config.GardenClusters {
+			kc := garden.KubeConfig
+			svr := getServerValueFromKubeconfig(kc)
+			if svr == serverName {
+				pgarden = garden.Name
+			}
+		}
+	}
+
+	if pgarden == "" {
+		fmt.Println("the server address enter could not be found as valid Garden target")
+		os.Exit(2)
+	}
+
+}
+
+func findIP(input string) string {
+	numBlock := "(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])"
+	regexPattern := numBlock + "\\." + numBlock + "\\." + numBlock + "\\." + numBlock
+
+	regEx := regexp.MustCompile(regexPattern)
+	return regEx.FindString(input)
+}
+
+func checkServerName(input string) bool {
+	if strings.HasPrefix(input, "https://") && strings.HasSuffix(input, ":443") {
+		return true
+	}
+	return false
+}
+
+func getServerValueFromKubeconfig(kubeconfigPath string) string {
+	var kc Kubeconfig
+	yamlFile, err := ioutil.ReadFile(kubeconfigPath)
+	checkError(err)
+	err = yaml.Unmarshal(yamlFile, &kc)
+	checkError(err)
+	if kc.Clusters[0].Cluster.Server == "" {
+		fmt.Println("can't get server value from kubeconfig")
+		os.Exit(2)
```

Comment:
not sure if the program should exit in this case or if it should just skip / ignore it in this case

Reply:
thanks, i removed 'os.Exit(2)'